src/ingest/src/main/scala/osmesa/ingest/ProcessOSM.scala

```diff
@@ -180,7 +184,7 @@ object ProcessOSM {
       .distinct
 
     val fullReferencedWays = allReferencedWays
-      .select('changeset, 'id, 'version, 'updated, 'visible, posexplode_outer('nds).as(Seq("idx", "ref")))
+      .select('changeset, 'id, 'version, 'updated, 'visible, posexplode('nds).as(Seq("idx", "ref")))
```

Comment:
This actually does need to be `posexplode_outer` (though that does force a dependency on Spark-2.2.0+); without it, deleted (`visible == false`) elements expand to 0 rows (vs. 1 matched w/ `null`s), removing them from subsequent processing (and causing the eventual `validUntil` to become invalid, since the set of reconstructed geometries will only include visible versions).

Reply:
Very good.  The spark documentation that I found didn't clearly declare the difference between the two versions.  I'll fix this.